oftraf.py

```diff
@@ -136,16 +154,26 @@ def of_sniff(ifname, ofport):
     of13_out_pkts_malformed = 0L
 
     try:
-        for _, pkt in pcap.pcap(name=ifname, immediate=True):
+        for _, pkt in pcap.pcap(name=ifname, immediate=False):
 
             eth = dpkt.ethernet.Ethernet(pkt)
-            tcp = eth.data.data
-            nbytes = len(eth.data)
+            if hasattr(eth.data, 'data'):
+                tcp = eth.data.data
+            else:
+                continue
 
             if type(tcp) != dpkt.tcp.TCP:
                 tcp_pkts_malformed += 1
                 continue
 
+            # do not further analyze the packet if it does not have any
+            # OpenFlow payload
+            payload = tcp.data
+            if len(payload) <= 1:
```

Comment:
This cannot be changed because payload is reavaluated and its length changes between the repeats of the while statement in lines 192-198